### PR TITLE
Helpful TypeError when user forgets to return `values` in root valida…

### PR DESCRIPTION
**Pull request by masalim2**
## Change Summary

If a user forgot to return the `values` dict in a root validator, they'd see a somewhat opaque exception: 
```
TypeError: __dict__ must be set to a dictionary, not a 'NoneType'`.  
```

Added a quick check in `BaseModel.__init__` to raise a more user-friendly hint in the TypeError:
```
TypeError: BaseModel.__init__ received `None` values. Are you forgetting to return `values` in a root validator?
```

## Checklist

* [X] Unit tests for the changes exist
* [X] Tests pass on CI and coverage remains at 100%
* [X] Documentation reflects the changes where applicable
* [X] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)

### CODE REVIEW

File: pydantic/main.py [hunk 1]
```diff
@@ -360,11 +360,15 @@ def __init__(__pydantic_self__, **data: Any) -> None:
         values, fields_set, validation_error = validate_model(__pydantic_self__.__class__, data)
         if validation_error:
             raise validation_error
+        if values is None:
```
Comment (PrettyWood) [inline pydantic/main.py:363]: Good idea! 👍 
But what if you do `return 3` in your `root_validator` you will still have an implicit error `__dict__ must be set to a dictionary, not a 'int'`! Maybe we could handle a more generic case with something like
```python
try:
    object_setattr(__pydantic_self__, '__dict__', values)
except TypeError as e:
    raise TypeError('Did you return a valid dict in your root validators ?') from e
```
?
Just my 2 cents :)

Comment (masalim2) [reply pydantic/main.py]: Thanks for the feedback!  This looks good -- updated the PR 👍

File: pydantic/main.py [hunk 2]
```diff
@@ -360,11 +360,14 @@ def __init__(__pydantic_self__, **data: Any) -> None:
...
-        object_setattr(__pydantic_self__, '__dict__', values)
+        try:
+            object_setattr(__pydantic_self__, '__dict__', values)
+        except TypeError as e:
+            raise TypeError('Did you return a valid dict in your root validators?') from e
         object_setattr(__pydantic_self__, '__fields_set__', fields_set)
         __pydantic_self__._init_private_attributes()
 
-    @no_type_check
+    @no_type_check  # noqa: C901 (ignore complexity)
```
Comment (samuelcolvin) [inline pydantic/main.py:370]: this looks unrelated.

File: pydantic/main.py [hunk 3]
```diff
@@ -360,11 +360,14 @@ def __init__(__pydantic_self__, **data: Any) -> None:
         values, fields_set, validation_error = validate_model(__pydantic_self__.__class__, data)
         if validation_error:
             raise validation_error
-        object_setattr(__pydantic_self__, '__dict__', values)
+        try:
+            object_setattr(__pydantic_self__, '__dict__', values)
+        except TypeError as e:
+            raise TypeError('Did you return a valid dict in your root validators?') from e
```
Comment (samuelcolvin) [inline pydantic/main.py:366]: the exception message should not be a question, maybe something like

> model values must be a dict, you may not have returned a dictionary from a root validator

**Review by samuelcolvin — COMMENTED** [review]
otherwise LGTM.

File: pydantic/main.py (hunk 2 above)
Comment (PrettyWood) [reply pydantic/main.py:370]: It's because of https://github.com/samuelcolvin/pydantic/pull/2198#pullrequestreview-550921099

**Review by PrettyWood — COMMENTED** [review]
@masalim2 Could you change the exception message?

File: changes/2209-masalim2.md [hunk 4]
```diff
@@ -0,0 +1 @@
+User-friendly TypeError when a `root_validator` returns `None`.
```
Comment (PrettyWood) [inline changes/2209-masalim2.md:1]: Better add a verb at the beginning like
> Raise a user-friendly `TypeError` when a `root_validator` does not return a dict (e.g. `None`).

Otherwise LGTM 👍

**Review by PrettyWood — APPROVED** [review]
